src/model/encoding/__tests__/convertFromHTMLToContentBlocks-test.js

```diff
@@ -54,7 +43,83 @@ const testConvertingAdjacentHtmlElementsToContentBlocks = (tag: string) => {
   'li',
   'p',
   'pre',
-].forEach(tag => testConvertingAdjacentHtmlElementsToContentBlocks(tag));
+];
+
+const assertConvertFromHTMLToContentBlocks = (html_string, config = {}) => {
+  const options = {
+    ...DEFAULT_CONFIG,
+    ...config,
+  };
+
+  const {DOMBuilder, blockRenderMap, experimentalTreeDataSupport} = options;
+
+  expect(
+    convertFromHTMLToContentBlocks(
+      html_string,
+      DOMBuilder,
+      blockRenderMap,
+      experimentalTreeDataSupport,
+    ).contentBlocks.map(block => block.toJS()),
+  ).toMatchSnapshot();
+};
+
+const testConvertingAdjacentHtmlElementsToContentBlocks = (
+  tag: string,
+  experimentalTreeDataSupport?: boolean = false,
+) => {
+  test(`must not merge tags when converting adjacent <${tag} />`, () => {
```

Comment:
I love this. Sometimes it's better to be repetetive in tests, but I think this is DRY while still being really clear.